gitoxide-core/src/hours/util.rs

```diff
@@ -43,6 +42,29 @@ impl<'a> From<&'a WorkByEmail> for WorkByPerson {
     }
 }
 
+fn join<I>(mut iter: I, sep: &str) -> String
```

Comment:
IMO you can avoid the `String` creation by turning this into a struct, with `iter` and `sep` stored in it.

Then impl `fmt::Display` on it, and you will achieve zero-allocation joining.

Reply:
I think it's fine to allocate here, it's similar to what was available before with `std::slice::join()`.

Reply:
I took it from here https://github.com/rust-itertools/itertools/blob/762643f1be2217140a972745cf4d6ed69435f722/src/lib.rs#L2295-L2324

I also think its fine as is but one could certainly think of ways to improve it. The function could also be made non generic and just accept a slice as that would have less indirections in this case but I figured the way it is now it is easier to move it to utils in case it happens to be needed somewhere else.

Reply:
Oh, if you find the time, maybe even in the other PR, could you add a note to say where it's from?
I'd like to keep track of any code that isn't specifically written for `gitoxide` to be sure I can comply with licensing terms.
Thanks in advance.

Reply:
yeah sure 👍